src/ctapipe/calib/camera/calibrator.py

```diff
@@ -282,6 +293,44 @@ def _calibrate_dl1(self, event, tel_id):
         # store the results in the event structure
         event.dl1.tel[tel_id] = dl1
 
+    def _process_calib(self, event, tel_id):
+        waveforms = event.r1.tel[tel_id].waveform
+        if self._check_r1_empty(waveforms):
+            return
+
+        if self.image_extractor_type.tel[tel_id] not in [
```

Comment:
Why do you only allow these two extractors? We have an abstract API for a reason to enable users to chose.

Reply:
The check has been removed.